lib/vagrant-proxyconf/action/configure_env_proxy.rb

```diff
@@ -61,6 +63,40 @@ def set_windows_system_proxy(proxy)
           end
         end
 
+        def set_windows_auto_config(autoconfig)
+          if autoconfig
+            path    = "HKCU\\Software\\Microsoft\\Windows\\CurrentVersion\\Internet Settings"
+            proxy1  = "cmd.exe /C reg add \"#{path}\" /v AutoConfigURL /t REG_SZ    /d #{config.autoconfig.inspect} /f"
+
+            logger.info('Setting system auto config settings')
+
+            @machine.communicate.sudo(proxy1)
+
+            set_windows_ie_settings
+
+          else
+            logger.info("Not setting auto config settings")
+          end
+        end
+
+        def set_windows_ie_settings
+          path    = "HKCU\\Software\\Microsoft\\Windows\\CurrentVersion\\Internet Settings\\Connections"
+          keys = ["DefaultConnectionSettings", "SavedLegacySettings"]
+
+          keys.each {
+            |key| connectionHex1 = "cmd.exe /C reg query \"#{path}\" /v #{key} /t REG_BINARY"
+            @machine.communicate.sudo(connectionHex1) do |type, data|
+              if type == :stdout
+                if data.include? key
+                  hex = update_hex(data.split()[2], 8, "05")
```

Comment:
This is a bit to magic with those parameters. Maybe just pass the data to the method, and possibly rename the method to be more descriptive.